### ISSUE TITLE
Symbolic Links Aren't Created When Needed Using TF_SYSTEM_LIBS

### ISSUE THREAD

**Issue by Amzo** [tensorflow/tensorflow#31020]
**System information**
- OS: FreeBSD
- Building from source 
- TensorFlow version: V1.14.0
- Python version: 3.6
- Bazel version (if compiling from source):  0.28
- GCC/Compiler version (if compiling from source): Clang 6.0.1

**Describe the problem**

When using _TF_SYSTEM_LIBS_ to build Tensorflow from source. It tries to include files which it has yet to create. In the file third_party/systemlibs/protobuf.BUILD. It creates symbolic links to the header files when building with _TF_SYSTEM_LIBS_. However it tries to compile code which relies on these headers before these headers have been created. If I continue the build without cleaning the build continues as normal. 

Setting jobs to 1 even causes this problem. As the files either haven't been created or haven't finished creating them. I can't tell if this is an issue with Bazel or Tensorflow, but it's either not waiting for the sub command to complete and create these files, or when using _TF_SYSTEM_LIBS_ it is compiling code which relies on these before they're created.

**Provide the exact sequence of commands / steps that you executed before running into the problem**

**Compile Tensorflow with the following _TF_SYSTEM_LIBS:_**

```
absl_py astor_archive 
boringssl
com_github_googleapis_googleapis
com_github_googlecloudplatform_google_cloud_cpp
com_google_protobuf
com_google_protobuf_cc
com_googlesource_code_re2
curl
cython
double_conversion
enum34_archive
flatbuffers
gast_archive
gif_archive grpc
hwloc
icu
jpeg
jsoncpp_git
keras_applications_archive
lmdb
nasm
nsync org_sqlite
pasta
pcre
png_archive
protobuf_archive
six_archive
snappy
swig
termcolor_archive
wrapt
zlib_archive"
```

**Additional enviroment variables for building:**

```
PYTHON_BIN_PATH=${PYTHON_CMD}"
PYTHON_LIB_PATH="${PYTHON_SITELIBDIR}"
TF_NEED_JEMALLOC=0
TF_NEED_KAFKA=0
TF_NEED_OPENCL_SYCL=0
TF_NEED_AWS=0
TF_NEED_GCP=0
TF_NEED_HDFS=0
TF_NEED_S3=0
TF_ENABLE_XLA=0
TF_NEED_GDR=0
TF_NEED_VERBS=0
TF_NEED_OPENCL=0
TF_NEED_MPI=0
TF_NEED_TENSORRT=0
TF_NEED_NGRAPH=0
TF_NEED_IGNITE=0
TF_NEED_ROCM=0
TF_NEED_CUDA=0
TF_SET_ANDROID_WORKSPACE=0
TF_DOWNLOAD_CLANG=0
TF_NEED_NCCL=0
TF_NEED_OPENCL=0
TF_IGNORE_MAX_BAZEL_VERSION=1
CC_OPT_FLAGS="-march=${CPU_TARGET}"
PREFIX="${LOCALBASE}"
```

**Using the follow bazelrc file:**

```
startup --batch

build --strip=never
build --verbose_failures --noshow_loading_progress
test --verbose_test_summary --verbose_failures --noshow_loading_progress

build --spawn_strategy=local --genrule_strategy=local
test --spawn_strategy=local --genrule_strategy=local

fetch --repository_cache="%%BAZEL_DIR%%/bazel-cache/" --distdir="%%BAZEL_DIST%%/bazel-distdir/"
build --repository_cache="%%BAZEL_DIR%%/bazel-cache/" --distdir="%%BAZEL_DIST%%/bazel-distdir/"

build --define=PREFIX=%%LOCALBASE%%
build --define=LIBDIR=%%LOCALBASE%%/lib

build --config=noaws --config=nohdfs --config=noignite --config=nokafka
```

** Build Command Used**

```
bazel --bazelrc="${WRKDIR}/bazelrc" ${BAZEL_BOOT} build ${BAZEL_COPT} --host_copt="-I${LOCALBASE}/include" \
		--host_linkopt="-L${LOCALBASE}/lib" --linkopt="-L${LOCALBASE}/lib" --config=opt \
		--incompatible_no_support_tools_in_action_inputs=false \
		--verbose_failures -s \
		//tensorflow:libtensorflow.so \
		//tensorflow:libtensorflow_cc.so \
		//tensorflow:install_headers \
		//tensorflow/tools/pip_package:build_pip_package
```

**Any other info / logs**

Compilation log:

```
SUBCOMMAND: # //tensorflow/core:conv_autotuning_proto_cc_genproto [action 'ProtoCompile tensorflow/core/protobuf/conv_autotuning
.pb.cc [for host]']
(cd /usr/home/Amzo/FreeBSD-Tensorflow/science/py-tensorflow/work-py36/bazel_out/f19d4b0a99d0cb4c4a90e3adec2fef45/execroot/org_te
nsorflow && \
  exec env - \
    PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin:/home/Amzo/bin \
  bazel-out/host/bin/external/protobuf_archive/protoc.bin '--cpp_out=bazel-out/host/bin/' -I. -I. -Iexternal/protobuf_archive -I
bazel-out/host/bin/external/protobuf_archive -Iexternal/protobuf_archive -Ibazel-out/host/bin/external/protobuf_archive tensorfl
ow/core/protobuf/conv_autotuning.proto)
SUBCOMMAND: # //tensorflow/core:autotuning_proto_cc_genproto [action 'ProtoCompile tensorflow/core/protobuf/autotuning.pb.cc [fo
r host]']
(cd /usr/home/Amzo/FreeBSD-Tensorflow/science/py-tensorflow/work-py36/bazel_out/f19d4b0a99d0cb4c4a90e3adec2fef45/execroot/org_te
nsorflow && \
  exec env - \
    PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin:/home/Amzo/bin \
  bazel-out/host/bin/external/protobuf_archive/protoc.bin '--cpp_out=bazel-out/host/bin/' -I. -Iexternal/protobuf_archive -Ibaze
l-out/host/bin/external/protobuf_archive tensorflow/core/protobuf/autotuning.proto)
ERROR: /usr/home/Amzo/FreeBSD-Tensorflow/science/py-tensorflow/work-py36/tensorflow-1.14.0/tensorflow/core/BUILD:2255:1: ProtoCo
mpile tensorflow/core/protobuf/autotuning.pb.cc failed (Exit 1)
google/protobuf/any.proto: File not found.
google/protobuf/duration.proto: File not found.
tensorflow/core/protobuf/autotuning.proto:10:1: Import "google/protobuf/any.proto" was not found or had errors.
tensorflow/core/protobuf/autotuning.proto:11:1: Import "google/protobuf/duration.proto" was not found or had errors.
tensorflow/core/protobuf/autotuning.proto:52:3: "google.protobuf.Duration" is not defined.
```

While I know FreeBSD isn't a supported platform, I just need help finding the culprit to this issue. As the problem only occasionally occurs. If I create a new jail and build, then sometimes it will succeed and sometimes it will fail due to trying to compile source before the proto headers have been linked. This is just based on luck as to whether or not another job has created them.

**Comment by gunan** [comment]
@perfinion as he helped set up `TF_SYSTEM_LIBS`

**Comment by Amzo** [comment]
I've just worked around the issue for now and patched the bzl file to just use the include files directly from ${LOCALBASE}/includes instead of just creating symlinks to the build directory.

**Comment by Flamefire** [comment]
I just ran into the same issue: `TF_SYSTEM_LIBS=com_google_protobuf` set but compile throws a "dangling symbolic link" to the protobuf headers. I was using the 2.0.0 tag

**Comment by Flamefire** [comment]
Actually it is not the same issue. The link is indeed dangling because TF uses `$(INCLUDEDIR)`. We have protobuf installed to a different prefix as other software (think `/opt/protobuf-x.y.z`) to isolate it properly. So what we need is something like `TF_PROTOBUF_PREFIX` to set this to the folder containing the `include`, `bin` etc folders.

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this, as you are using an older version of tensorflow which is officially considered end of life . We recommend that you upgrade to the latest 2.x version and let us know if the issue still persists in newer versions. Please open a new issue for any help you need against 2.x, and we will get you the right help. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, please provide us with more information.

**Comment by surak** [comment]
@Flamefire was this ever fixed? I have it on Tensorflow 2.5.0

**Comment by Flamefire** [comment]
@surak What exactly do you have? TF now uses another variable for the protobuf headers: https://github.com/tensorflow/tensorflow/blob/c313c925f4d525aef40914413299706074fb51f8/third_party/systemlibs/protobuf.BUILD#L57

This works so far in EasyBuild
